fennel/datasets/datasets.py

```diff
@@ -241,11 +245,50 @@ def transform(self, func: Callable, schema: Dict = {}) -> _Node:
             return Transform(self, func, None)
         return Transform(self, func, copy.deepcopy(schema))
 
-    def filter(self, func: Callable) -> _Node:
+    def filter(self, func: Callable | Expr) -> _Node:
         return Filter(self, func)
 
-    def assign(self, name: str, dtype: Type, func: Callable) -> _Node:
-        return Assign(self, name, dtype, func)
+    def assign(self, *args, **kwargs) -> _Node:
+        """
+        Assign is an overloaded operator that can be used in several ways:
+        1. Assigning a new column with a lambda function:
+            >>> ds.assign("new_column", int, lambda x: x.old_column + 1)
+        2. Using kwarg syntax to assign a column:
+            >>> ds.assign(new_column=F("old_column") + 1).astype(int))
+        3. Assigning multiple columns:
+            >>> ds.assign(
+            ...     new_column1=(F("old_column1") + 1).astype(int),
+            ...     new_column2=(F("old_column2") + 2).astype(int),
+            ... )
+
```

Comment:
2 and 3 feel the same so I wonder if we can simplify the docstring a bit. Also, do we support the approach where it's 3 kwarg but they provide expression instead of lambda?

Reply:
Yeah the former is args and if they do indeed use kwarg we typecheck. But looks like you suggested to remove the latter